pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -106,9 +106,89 @@ func (r *Reader) ReadToEnd(ctx context.Context) {
 		}
 	}()
 
+	doneReadingFile := r.readHeader(ctx)
+	if doneReadingFile {
+		return
+	}
+
+	// Reset position in file to r.Offest after the header scanner might have moved it past a content token.
+	if _, err := r.file.Seek(r.Offset, 0); err != nil {
```

Comment:
I don't think this should be in the hot path. Let's only do it once, when we detect we are done reading the header.

Reply:
Right, not sure what I was thinking leaving this call here. Moved to `readHeader` in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35870/commits/0f55d1cf41497186306df3e1f96e5aa98908e5d3.